packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,204 @@
+/// @title LivenessModule2
+/// @notice This module allows for challenge-based ownership transfer to a fallback owner
+///         when the Safe becomes unresponsive. The fallback owner can initiate a challenge,
+///         and if the Safe doesn't respond within the challenge period, ownership transfers
+///         to the fallback owner.
+/// @dev This is a singleton contract. To use it:
+///      1. The Safe must first enable this module using ModuleManager.enableModule()
+///      2. The Safe must then configure the module by calling enableModule() with parameters
+contract LivenessModule2 is ILivenessModule2 {
+    /// @notice Reserved address used as the previous owner to the first owner in a Safe
+    address public constant SENTINEL_OWNER = address(0x1);
+
+    /// @notice Configuration for a Safe's liveness module
+    struct ModuleConfig {
+        uint256 livenessResponsePeriod;
+        address fallbackOwner;
+    }
+
+    /// @notice Mapping from Safe address to its configuration
+    mapping(address => ModuleConfig) public safeConfigs;
+
+    /// @notice Mapping from Safe address to active challenge start time (0 if no challenge)
+    mapping(address => uint256) public challengeStartTime;
+
+    /// @notice Semantic version.
+    /// @custom:semver 2.0.0
+    string public constant version = "2.0.0";
+
+    /// @notice Enables the module by the multisig to be challenged and sets the liveness_response_period and
+    /// fallback_owner
+    /// @dev MUST set the caller as a safe
+    /// @dev MUST take as parameters liveness_response_period and fallback_owner and store them as related to the safe
+    /// @dev MUST accept an arbitrary number of independent safe contracts to enable the module
+    /// @param _livenessResponsePeriod The period in seconds for a liveness response
+    /// @param _fallbackOwner The address that will become owner if challenge succeeds
+    function enableModule(uint256 _livenessResponsePeriod, address _fallbackOwner) external {
+        if (_livenessResponsePeriod == 0 || _fallbackOwner == address(0)) {
+            revert LivenessModule2_InvalidParameters();
+        }
+
+        // Set the caller as a safe and store its configuration
+        ModuleConfig storage config = safeConfigs[msg.sender];
+
+        // Store the parameters related to this safe
+        config.livenessResponsePeriod = _livenessResponsePeriod;
+        config.fallbackOwner = _fallbackOwner;
+
+        // Clear any existing challenge when enabling/re-enabling
+        delete challengeStartTime[msg.sender];
+
+        emit ModuleEnabled(msg.sender, _livenessResponsePeriod, _fallbackOwner);
+    }
+
+    /// @notice Disables the module by an enabled safe
+    /// @dev MUST only be executable by an enabled safe
+    /// @dev MUST erase the existing liveness_response_period and fallback_owner data related to the calling safe
+    /// @dev Note: Disabling the module also cancels any ongoing challenges
+    function disable() external {
```

Comment:
```suggestion
    function clear() external {
```

Reply:
Fixed in [37ae803](https://github.com/ethereum-optimism/optimism/pull/17272/commits/37ae803b8080e1b2aa502cd33d3a24d3e0134ed2)